corehq/apps/api/migrations/0003_rename_apiuser.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.27 on 2020-02-08 03:22
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```

Comment:
I am not sure whether it is easier overall, but I think we could avoid this migration and end up in the same place. Instead of creating a SQLFoo model in corehq/apps/foo_app/models.py, and then renaming it later, what if in step 1 of the migration we move the Couch model from corehq/apps/foo_app/models.py to corehq/apps/foo_app/couch_models.py. A full-featured IDE should make this relatively easy, and changes should just be in import statements. Add the SQL model (with the same name as the Couch model) to corehq/apps/foo_app/models.py.

Would that be easier? Are there risks involved in moving the Couch model?

Reply:
That would be easier, one fewer django migration would mean fewer merge conflicts.

I've always thought there was a magic link between couch class names and doc types. Some classes have a `base_doc` attribute that seems like it might be an override for doc type. Do you know anything about that?

Reply:
Some python meta magic involved in the doc_type attribute but looks like it's just a normal JsonObject field: https://github.com/dimagi/couchdbkit/blob/09919ebd7c809c7a114b1b18997665ae41be6d2e/couchdbkit/schema/base.py#L58

It may work to just override it @dannyroberts may know better

Reply:
I experimented with this and found a few places where we assume that doc_type is the class name, like [this](https://github.com/dimagi/commcare-hq/blob/4ea1fcf1a9659f3a2b69ddd6fb2069f1e3a64415/corehq/apps/domain/dbaccessors.py#L7). For now I'm going to stick with the current approach, prefixing the new model with `SQL` and then renaming it later.